File: FutsalSystemFront/src/app/Models/MatchEventDto.ts
import { PlayerEvent } from '../Enums/PlayerEvent';

export class MatchEventDto {
    public Id: number;
    public MatchId: number;
    public Minute: number;
    public PlayerId: number;
    public PlayerName: string;
    public TeamName: string;
    public EventType: PlayerEvent;

    constructor(id: number, matchId: number, minute: number, playerId: number, playerName: string, teamName: string, eventType: PlayerEvent) {
        this.Id = id;
        this.MatchId = matchId;
        this.Minute = minute;
        this.PlayerId = playerId;
        this.PlayerName = playerName;
        this.TeamName = teamName;
        this.EventType = eventType;
    }
}

File: FutsalSystem/FutsalSystem/Models/DTO/MatchEvent/MatchEventDTO.cs
using FutsalSystem.Base.Interface;
using FutsalSystem.Enumerators;
using System.ComponentModel.DataAnnotations.Schema;

namespace FutsalSystem.Models.DTO.MatchEvent
{
    public class MatchEventDTO : IBaseEntity
    {
        public int Id { get; set; }
        public int MatchId { get; set; }
        [ForeignKey("MatchId")]
        public Models.Match Match { get; set; }
        public int Minute { get; set; }
        public int PlayerId { get; set; }

        [ForeignKey("PlayerId")]
        public Models.Player Player { get; set; }
        public string PlayerName { get; set; }
        public string TeamName { get; set; }
        public PlayerEvent EventType { get; set; }
    }
}
